### PR TITLE
feat(onboarding button): [MAG-3048] extended button with rounded corners

### PR DESCRIPTION
**Pull request by eTheMan80**
# <MAG-3048 / Create button component>

## TL;DR

-   Created the Onboarding button component
-   Added variants to include disabled and loading state
-   
-  

## Details

Checklist / Reminders:

-   [x] Documentation / Story is created / adapted
-   [x] Unit tests
-   [x] CSS changes are BEM-compliant

Screenshot 2020-11-12 at 15 53 54
Screenshot 2020-11-12 at 15 53 47
Screenshot 2020-11-12 at 15 53 41

### CODE REVIEW

**Review by tricaalexandru — APPROVED** [review]

File: src/components/Atoms/Button/Button.interface.ts [hunk 1]
```diff
@@ -37,4 +39,6 @@ export interface ButtonProps
   round?: boolean
   squared?: boolean
   id?: string
+  rounded?: boolean
```
Comment (philjonasfiltered) [inline src/components/Atoms/Button/Button.interface.ts:42]: @eTheMan80 What is the difference between `round` and `rounded`? Also, could you please either add `rounded` to the `knobs` or add an example like we have with `Round` and `Squared`. Thanks 👍

File: src/components/Molecules/OnBoardingButton/OnBoardingButton.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React, { FC } from 'react'
+import { Button, ButtonProps, StyledButtonProps } from '@/components/Atoms/Button'
+
+export const OnBoardingButton: React.FC<ButtonProps> = ({
```
Comment (matthill8286) [inline src/components/Molecules/OnBoardingButton/OnBoardingButton.tsx:4]: @eTheMan80 The button needs it's own interface, and really doesn't need this many props, as it's an molecule it's got a specific use, so things like rounded, elevated would just be true, it doesn't need an actionType either as thats ghost, so the props would be

```
onboardingClick
to,
children
```

Interface would be

```
export interface OnboardingButtonProps {
    onboardingClick?: (ev: React.MouseEvent<HTMLElement>) => void
    to?: H.LocationDescriptor
    children?: React.ReactNode
}
```

File: src/components/Atoms/Button/Button.interface.ts (hunk 1 above)
Comment (eTheMan80) [reply src/components/Atoms/Button/Button.interface.ts:42]: @philjonasfiltered after speaking with @matthill8286 it was decided that the onboarding button component is an extension of the button and so doesn't need a separate example within the atom structure. The onboarding component is the example. I've updated the `rounded` prop to `curved`, does this sound better?

Comment (philjonasfiltered) [reply src/components/Atoms/Button/Button.interface.ts:42]: @eTheMan80 works for me 😄

**Review by philjonasfiltered — APPROVED** [review]

File: src/components/Molecules/OnBoardingButton/OnBoardingButton.tsx (hunk 2 above)
Comment (eTheMan80) [reply src/components/Molecules/OnBoardingButton/OnBoardingButton.tsx]: @matthill8286 this has been updated

**Review by matthill8286 — APPROVED** [review]

**Pull request merged**